client/src/containers/ProfilePage.jsx

```diff
@@ -87,23 +90,69 @@ const FormButton = styled('button', {
 
 export function ProfilePage() {
   const { user } = useAuth()
+  const { username, email, id, calorieLimit, image } = user
+
+  const [updatedUser, setUpdatedUser] = useState({
+    id,
+    username,
+    email,
+    calorieLimit,
+    image,
+  })
+  const [isEditing, setIsEditing] = useState(false)
+  const [isSubmitting, setIsSubmitting] = useState(false)
+
+  const { mutate: update } = useUpdateUser()
+
+  const handleSubmit = (e) => {
+    e.preventDefault()
+    setIsSubmitting(true)
+    return update(updatedUser)
+  }
+
+  const handleChange = (e) => {
+    return setUpdatedUser((prev) => ({ ...prev, [e.target.name]: e.target.value }))
+  }
 
   return (
     <ProfilePageWrapper>
       <Avatar>
-        <Picture />
+        <Picture>
+          <img src={image} alt="avatar" />
+        </Picture>
         <UserName>{user.username}</UserName>
       </Avatar>
       <UserDets>
         <UserDetailsHeading>User Info</UserDetailsHeading>
-        <Form>
-          <Label for="username">Username:</Label>
-          <Input name="userame" value={user.username} />
-          <Label for="email">Email:</Label>
-          <Input name="email" value={user.email} />
-          <Label for="calorieLimit">Calorielimit:</Label>
-          <Input name="calorieLimit" value="1000" />
-          <FormButton>Save</FormButton>
+        <Form onSubmit={handleSubmit}>
+          <Label htmlFor="username">Username:</Label>
+          <Input
+            name="username"
+            value={isEditing ? null : user.username}
```

Comment:
თუ არ ვაედიტებ null რატო ჩანს? ხომ უნდა ვხედავდე რა value არის ამჟამინდელი?

მე გირჩევ რომ აქ სულ ჩანდეს ველიუ და რომ დავაფოკუსებ რომელიმე ინფუთს მაშინ და-enable-დეს save ღილაკი. მანამდე disabled იყოს

ასევე მოგიწევს ლოგიკის დაწერა, რომლითაც უნდა შეამოწმო, ძველი უდრის თუ არა ახალ იუზერს. თუ არ უდრის მაშინ save is enabled, else disabled

Reply:
![snippet](https://user-images.githubusercontent.com/68153140/149004594-80f11cb9-d1e8-40d1-b2f9-be8e34634a11.png)

მისაღებია მსგავსი კოდი? User არის useAuth-იდან აღებული იუზერი, user კი ProfilePage-ის სტეიტში რაცაა და მისი შეცვლა(update) შეიძლება ინფუთებიდან ,თუ არაფერი შეიცვალა calls არ ვაკეთებ 

![snippet2](https://user-images.githubusercontent.com/68153140/149005031-bf47f644-ca81-4ef4-a805-167e0884883a.png)

Reply:
კარგია, მაგრამ naming საშინელია

useAuth არ დაადესტრუქტურირო.
`const auth = useAuth()` და მერე `auth.user.username === user.username`...

Reply:
საშინელი namingi User რომ არის  ეგ? ეხლა ჩავასწორე რაც მითხარი .auth.user.

 გოგა მაგალითად User-ის მაგივრად prevUser რომ დამეწერა,ეგეც ცუდი ნეიმინგი იქნებოდა?(ისე გეკითხები)

Reply:
არ იქნებოდა ცუდი.

იმიტომ გავაზვიადე შეფასება, რომ სახელში არ არის მანდ პრობლემა რეალურად. consistency ირღვევა. როცა ცვლადებს სახელს არქმევ cammelCase-თ და ერთგან მერე სხვანაირად გაქვს, ეგ ძალიან ცუდია.

User (Pascalcase) ან USER (UPPERCASE) გამოიყენება ხოლმე მუდმივებთან, ენამებთან.